### ISSUE TITLE
When sharing from another app, change the file name before tapping "upload"

### ISSUE THREAD

**Issue by FernandoMarques-Santos** [nextcloud/android#10259]
### Is your feature request related to a problem? Please describe.
Other apps generate random names (such as receipts), and in some cases, they include two dots ":" which create problems when syncing with Windows. Additionally, it would be nice for the workflow.

### Describe the solution you'd like
A box where you write the filename you want before hitting UPLOAD in the android app.

### Describe alternatives you've considered
I understand that Nextcloud may need to change the name only AFTER the app shared the content. So maybe this should be a feature that gives instructions to the server after the upload.

**Comment by joshtrichards** [comment]
Duplicate of #9215
